### PR TITLE
Added schema and resolver for product list by category

### PR DESCRIPTION
**Pull request by kalypso-byte**

### CODE REVIEW

File: category/interfaces/graphql/schema.graphql [hunk 1]
```diff
@@ -1,3 +1,20 @@
...
+    name: String!
+    path: String!
+    active: Boolean!
+    promoted: Boolean!
+}
+
+
+type Commerce_CategoryData implements Commerce_Category {
+    code: String!
+    name: String!
```
Comment (danielpoe) [inline category/interfaces/graphql/schema.graphql]: What is with the other attributes? Category can have "Media" and Attributes?

File: product/interfaces/graphql/schema.graphql [hunk 2]
```diff
@@ -145,6 +145,12 @@ type Commerce_ProductPriceInfo {
 
 }
 
+type Commerce_CategoryProductList {
+    category: Commerce_Category!
```
Comment (danielpoe) [inline product/interfaces/graphql/schema.graphql]: Until now the "product" modul didnt know about the "category" modul. It is the other way around.
Cool would be a concept to search products by "any" filter like we have in the domain model..

Current dependency is:

Category -> Search
Category (->) Product -> Search

According to this the following idea:
Search Modul:
- Offer a "Commerce_SearchFilter interface" 
- Offers "Commerce_SearchMeta"
- Offers also implementations - e.g. a "KeyValueSearchFilter" ...

Product Modul:

Commerce:ProductSearchResult {
 products: [Commerce_Product!]
 meta: Commerce_SearchMeta
}

offer a Query:
- Commerce_ProductSearch(Commerce_SearchFilter[])


Category:
- Offers a Commerce_CategoryFilter (implements Commerce_SearchFilter)

File: category/interfaces/graphql/schema.graphql [hunk 3]
```diff
@@ -1,3 +1,20 @@
...
+    path: String!
+    active: Boolean!
+    promoted: Boolean!
+}
+
 interface Commerce_Tree {
     code: String!
     name: String!
```
Comment (danielpoe) [inline category/interfaces/graphql/schema.graphql]: Commerce_CategorySearchResult {
 Category: Commerce_Category
 Product: ProductSearchResult
}

Query 
Commerce_Category(categorycode, Commerce_SearchFilter[]): Commerce_CategorySearchResult

Comment (kalypso-byte) [reply category/interfaces/graphql/schema.graphql]: Moved logic to category and added filter options

**Pull request merged**